django/db/models/query.py

```diff
@@ -1153,6 +1172,36 @@ def __init__(self, *args, **kwargs):
         raise TypeError("EmptyQuerySet can't be instantiated")
 
 
+class RestrictedQuerySet(object):
+    # TODO: Ideas on how to make that nicer?
+    _allowed_methods = set()
+
+    def __init__(self, qs):
+        self._qs = qs
+
+    def __repr__(self):
+        data = list(self[:REPR_OUTPUT_SIZE + 1])
+        if len(data) > REPR_OUTPUT_SIZE:
+            data[-1] = "...(remaining elements truncated)..."
+        return '<%s %r>' % (self.__class__.__name__, data)
+
+    def __getitem__(self, k):
+        return self.__getattr__('__getitem__')(k)
```

Comment:
is this necessary? doesn't `__getitem__` go through `__getattr__` already if it's not defined?